File: backend/csharp-aspnet-webapi/Features/Job/JobStep.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

[Table("job_step")]
public class JobStep
{
    [Key]
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    [Column("id")]
    public int Id { get; set; }

    [Column("job_id")]
    public int JobId { get; set; }

    [Required]
    [Column("step_name")]
    public required string StepName { get; set; }   // "Downloading video at 720p" etc.

    [Required]
    [Column("status")]
    public required JobStepStatus Status { get; set; }

    [Column("note")]
    public string? Note { get; set; }               // optional detail shown to user

    [Column("error_msg")]
    public string? ErrorMsg { get; set; }

    [Column("started_at")]
    public DateTime? StartedAt { get; set; }

    [Column("ended_at")]
    public DateTime? EndedAt { get; set; }
}

File: backend/java-springboot/src/main/java/com/lulo97/backend/features/job/jobstep/JobStep.java
package com.lulo97.backend.features.job.jobstep;

import jakarta.persistence.*;
import java.time.LocalDateTime;
import com.lulo97.backend.NoLockId;

@Entity
@Table(name = "job_step")
public class JobStep {

    @Id
    @NoLockId
    @Column(name = "id")
    private Long id;

    @Column(name = "job_id", nullable = false)
    private Long jobId;

    @Column(name = "step_name", nullable = false)
    private String stepName;

    @Enumerated(EnumType.STRING)
    @Column(name = "status", nullable = false)
    private JobStepStatus status;

    @Column(name = "note")
    private String note;

    @Column(name = "error_msg")
    private String errorMsg;

    @Column(name = "started_at")
    private LocalDateTime startedAt = LocalDateTime.now();

    @Column(name = "ended_at")
    private LocalDateTime endedAt;

    // Constructors
    public JobStep() {}

    // Getters and setters

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Long getJobId() {
        return jobId;
    }

    public void setJobId(Long jobId) {
        this.jobId = jobId;
    }

    public String getStepName() {
        return stepName;
    }

    public void setStepName(String stepName) {
        this.stepName = stepName;
    }

    public JobStepStatus getStatus() {
        return status;
    }

    public void setStatus(JobStepStatus status) {
        this.status = status;
    }

    public String getNote() {
        return note;
    }

    public void setNote(String note) {
        this.note = note;
    }

    public String getErrorMsg() {
        return errorMsg;
    }

    public void setErrorMsg(String errorMsg) {
        this.errorMsg = errorMsg;
    }

    public LocalDateTime getStartedAt() {
        return startedAt;
    }

    public void setStartedAt(LocalDateTime startedAt) {
        this.startedAt = startedAt;
    }

    public LocalDateTime getEndedAt() {
        return endedAt;
    }

    public void setEndedAt(LocalDateTime endedAt) {
        this.endedAt = endedAt;
    }
}
